DataManagementSystem/DB/FileCatalogComponents/FileMetadata.py

```diff
@@ -330,127 +330,98 @@ def __transformMetaParameterToData( self, metaname ):
     result = self.db._update( req )
     return result
 
-  def __createMetaSelection( self, meta, value, table = '' ):
-
-    if type( value ) == types.DictType:
-      selectList = []
+#########################################################################
+#
+#  Finding files by metadata
+#
+#########################################################################
+
+  def __createMetaSelection( self, value ):
+    ''' Create selection string to be used in the SQL query
+    '''
+    queryList = []
+    if type( value ) == types.FloatType:
```

Comment:
As a matter of fact, ypou may do something like isinstance(obj, basestring) to compare ALL string types.